File: examples/python/core/parallel/access-individual-results.py
from aws_durable_execution_sdk_python import (
    BatchResult,
    DurableContext,
    durable_execution,
)

@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    """Access individual results from parallel execution."""
    
    def task_a(ctx: DurableContext) -> str:
        return ctx.step(lambda _: "Result A")
    
    def task_b(ctx: DurableContext) -> str:
        return ctx.step(lambda _: "Result B")
    
    def task_c(ctx: DurableContext) -> str:
        return ctx.step(lambda _: "Result C")
    
    result: BatchResult[str] = context.parallel([task_a, task_b, task_c])
    
    results = result.get_results()
    
    # Access results by index
    first_result = results[0]   # "Result A"
    second_result = results[1]  # "Result B"
    third_result = results[2]   # "Result C"
    
    return {
        "first": first_result,
        "second": second_result,
        "third": third_result,
        "all": results,
    }

File: examples/java/core/parallel/access-individual-results.java
public class AccessResultsHandler extends DurableHandler<Map<String, Object>, Map<String, Object>> {

    @Override
    public Map<String, Object> handleRequest(Map<String, Object> event, DurableContext ctx) {
        // Launch three async child contexts
        var futureA = ctx.runInChildContextAsync("task-a", String.class,
            child -> child.step("step-a", String.class, stepCtx -> "Result A"));

        var futureB = ctx.runInChildContextAsync("task-b", String.class,
            child -> child.step("step-b", String.class, stepCtx -> "Result B"));

        var futureC = ctx.runInChildContextAsync("task-c", String.class,
            child -> child.step("step-c", String.class, stepCtx -> "Result C"));

        // Wait for all to complete
        DurableFuture.allOf(futureA, futureB, futureC);

        // Access results by index
        String first = futureA.get();   // "Result A"
        String second = futureB.get();  // "Result B"
        String third = futureC.get();   // "Result C"

        return Map.of(
            "first", first,
            "second", second,
            "third", third,
            "all", List.of(first, second, third)
        );
    }
}
